File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
       int sum=0;
       int c=0;
       float avg=0;
       for(int i=0;i<k;i++){
        sum+=arr[i];
       } 
       avg=(float)sum/k;
        if(avg>=threshold)
           c++;
       for(int j=k;j<arr.length;j++){
        sum=sum-arr[j-k]+arr[j];
        avg=(float)sum/k;
        if(avg>=threshold)
           c++;
       }
      return c; 
    }
}

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        s=0
        c=0
        ws=0
        avg=0
        for i in range(k):
           s+=arr[i]
        if(s/float(k)>=threshold):
            c+=1
        ws=s
        for j in range(k,len(arr)):
            ws=ws-arr[j-k]+arr[j]
            if(ws/float(k)>=threshold):
                c+=1
        return c
